specs/governance/governor.md

```diff
@@ -0,0 +1,702 @@
+    - [`setVotingDelay`](#setvotingdelay)
+    - [`setVotingPeriod`](#setvotingperiod)
+    - [`updateQuorumNumerator`](#updatequorumnumerator)
+  - [Getters](#getters)
+    - [`approvedModules`](#approvedmodules)
+    - [`getProposalType`](#getproposaltype)
+    - [`getVotes`](#getvotes)
+    - [`getVotesWithParams`](#getvoteswithparams)
+    - [`hasVoted`](#hasvoted)
+    - [`hashProposal`](#hashproposal)
+    - [`hashProposalWithModule`](#hashproposalwithmodule)
+    - [`manager`](#manager)
+    - [`name`](#name)
+    - [`proposalDeadline`](#proposaldeadline)
+    - [`proposalSnapshot`](#proposalsnapshot)
+    - [`proposalThreshold`](#proposalthreshold)
+    - [`proposalVotes`](#proposalvotes)
+    - [`quorum`](#quorum)
+    - [`quorumDenominator`](#quorumdenominator)
+    - [`quorumNumerator`](#quorumnumerator)
+    - [`quorumNumerator` (block)](#quorumnumerator-block)
+    - [`state`](#state)
+    - [`supportsInterface`](#supportsinterface)
+    - [`token`](#token)
+    - [`version`](#version)
+    - [`votableSupply`](#votablesupply)
+    - [`votableSupply` (block)](#votablesupply-block)
+    - [`votingDelay`](#votingdelay)
+    - [`votingPeriod`](#votingperiod)
+    - [`weightCast`](#weightcast)
+  - [Events](#events)
+    - [`ProposalCanceled`](#proposalcanceled)
+    - [`VoteCast`](#votecast)
+    - [`VoteCastWithParams`](#votecastwithparams)
+    - [`ProposalTypeUpdated`](#proposaltypeupdated)
+    - [`ProposalExecuted`](#proposalexecuted)
+    - [`ProposalCreated`](#proposalcreated)
+    - [`ProposalDeadlineUpdated`](#proposaldeadlineupdated)
+    - [`ProposalThresholdSet`](#proposalthresholdset)
+    - [`VotingDelaySet`](#votingdelayset)
+    - [`VotingPeriodSet`](#votingperiodset)
+    - [`QuorumNumeratorUpdated`](#quorumnumeratorupdated)
+- [Storage](#storage)
+- [Types](#types)
+  - [`ProposalVote`](#proposalvote)
+  - [`ProposalCore`](#proposalcore)
+- [Proposal Types](#proposal-types)
+- [Modules](#modules)
+- [Security Considerations](#security-considerations)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The `Governor` contract implements the core governance logic for creating, voting, and executing proposals.
+This contract uses the `GovernanceToken` contract for voting power snapshots, and the `ProposalTypesConfigurator`
+for proposal types.
+
+## Constants
+
+| Constant          | Value                           | Description |
+| ----------------- | ------------------------------- | ----------- |
+| `BALLOT_TYPEHASH` |  `0x150214d74d59b7d1e90c73fc22ef3d991dd0a76b046543d4d80ab92d2a50328f` | The EIP-712 typehash for the ballot struct  |
+| `EXTENDED_BALLOT_TYPEHASH` |  `0x7f08F3095530B67CdF8466B7a923607944136Df0` | The EIP-712 typehash for the extended ballot struct |
+| `COUNTING_MODE` | `support=bravo&quorum=against,for,abstain&params=modules` | The configuration for supported values for `castVote` and how votes are counted |
```

Comment:
I am confused why this is a query string, i don't see `COUNTING_MODE` anywhere else in this doc

Reply:
should be used in the `castVote` function, updated in https://github.com/ethereum-optimism/specs/pull/315/commits/8d792d2c73a4063d0172225e231e5dfd5e682fcc